app/jobs/add_workflow_job.rb

```diff
@@ -2,26 +2,29 @@
 
 ##
 # Job to add a workflow to and object
-class AddWorkflowJob < GenericJob
-  attr_reader :workflow_name
+class AddWorkflowJob < BulkActionJob
+  def workflow_name
+    params.fetch(:workflow)
+  end
+
+  class BulkActionItem < BulkActionItem
+    delegate :workflow_name, to: :job
+
+    def perform
+      return unless check_update_ability?
```

Comment:
BulkActionItem provides re-usable methods like success!, failure!, and check_update_ability?